🔀 메인 화면 퍼블리싱

leehyeonbin opened the pull request.

leehyeonbin committed:
기본 로직 제거

leehyeonbin committed:
Coil 의존성 추가

Coil 라이브러리를 gradle/libs.versions.toml 파일과 composeApp/build.gradle.kts 파일에 추가했습니다.

leehyeonbin committed:
세팅 아이콘 추가

leehyeonbin committed:
✨ 홈 화면 퍼블리싱

khs3994 commented on composeApp/src/commonMain/kotlin/com/msg/onmip/feature/main/components/ColorRecommendationComponent.kt:
```diff
@@ -0,0 +1,58 @@
+            Text("파랑", style = TextStyle(color = Black, fontSize = 16.sp, fontFamily = AppFont()))
+        }
+        Box(modifier = Modifier.width(16.dp))
+        Column(modifier = Modifier.weight(1f), horizontalAlignment = Alignment.CenterHorizontally) {
+            Box(
+                modifier = Modifier.fillMaxWidth().aspectRatio(1f)
+                    .clip(shape = RoundedCornerShape(12.dp)).background(color = Yellow)
+            )
+            Box(modifier = Modifier.height(16.dp))
+            Text("노랑", style = TextStyle(color = Black, fontSize = 16.sp, fontFamily = AppFont()))
+        }
```
색상 추천 아이템도 컴포넌트화 하는건 어떨까요?

leehyeonbin replied:
컴포넌트화 하려다가 굳이 싶어서 안했는데 좋습니다~

khs3994 commented on composeApp/src/commonMain/kotlin/com/msg/onmip/feature/main/components/ColorRecommendationComponent.kt:
```diff
@@ -0,0 +1,58 @@
+            Box(modifier = Modifier.height(16.dp))
+            Text("노랑", style = TextStyle(color = Black, fontSize = 16.sp, fontFamily = AppFont()))
+        }
+        Box(modifier = Modifier.width(16.dp))
```
Spacer 대신 Box 사용하신 이유가 있을까요?
Row에 spacedBy로도 대체할 수 있을 것 같습니다!

leehyeonbin replied:
Flutter랑 React로 기억이 좀 섞여서 생각나는 컴포넌트가 몇 없어서 Box로 했습니다. 생각해보니 Spacer라는게 있었죠 감사합니당
spacedBy도 좋습니다
![image](https://github.com/user-attachments/assets/4f472784-d035-4f64-b3f4-77c65fc3a2e5)

khs3994 commented:
컴포넌트 패딩같은 경우에 screen 레벨에서 modifier로 넘겨주는건 어떻게 생각하시나요?
ex)
```kotlin
        MainAppBar(modifier = Modifier.padding(,,,))
        ListHeader(
           modifier = Modifier.padding(,,,),
           title = "날씨"
        )
        WeatherComponent(modifier = Modifier.padding(,,,))
        ListHeader(
           modifier = Modifier.padding(,,,),
           title = "추천 색상"
        )
        ColorRecommendationComponent(modifier = Modifier.padding(,,,))
        ListHeader(
           modifier = Modifier.padding(,,,),
           title = "추천"
        )
        OutfitRecommendationComponent(modifier = Modifier.padding(,,,))
```
패딩을 외부에서 지정해주는게 여러 경우에서 재사용하기 편했던 경험이 있습니다..!
여러 화면에서 전부 사용하지 않고 메인 스크린에서만 사용할 컴포넌트라면 안쪽에 패딩이 있어도 상관은 없어보입니다.
의견일 뿐이라 참고만 부탁드립니다

leehyeonbin commented:
> 컴포넌트 패딩같은 경우에 screen 레벨에서 modifier로 넘겨주는건 어떻게 생각하시나요? ex)
> 
> ```kotlin
>         MainAppBar(modifier = Modifier.padding(,,,))
>         ListHeader(
>            modifier = Modifier.padding(,,,),
>            title = "날씨"
>         )
>         WeatherComponent(modifier = Modifier.padding(,,,))
>         ListHeader(
>            modifier = Modifier.padding(,,,),
>            title = "추천 색상"
>         )
>         ColorRecommendationComponent(modifier = Modifier.padding(,,,))
>         ListHeader(
>            modifier = Modifier.padding(,,,),
>            title = "추천"
>         )
>         OutfitRecommendationComponent(modifier = Modifier.padding(,,,))
> ```
> 
> 패딩을 외부에서 지정해주는게 여러 경우에서 재사용하기 편했던 경험이 있습니다..! 여러 화면에서 전부 사용하지 않고 메인 스크린에서만 사용할 컴포넌트라면 안쪽에 패딩이 있어도 상관은 없어보입니다. 의견일 뿐이라 참고만 부탁드립니다

딱히 생각 안해봣는데 이후에 재사용할 거 생각하면 padding을 외부에서 추가해주는 편이 좀 더 좋아보이네요 감사합니다~

leehyeonbin committed:
옷, 색상 추천 컴포넌트 추가

옷, 색상 추천 컴포넌트를 추가합니다.

leehyeonbin committed:
ColorRecommendationItem 컴포넌트 내부 배치 수정

ColorRecommendationItem 컴포넌트의 Column 내부 요소들의 세로 간격을 16dp로 설정하고, 불필요한 Spacer를 제거했습니다.

leehyeonbin committed:
전체적인 패딩 조절

leehyeonbin committed:
feature/main 화면 구현

- 플랫폼별 화면 연결
- ColorRecommendationComponent 수정

leehyeonbin committed:
safeContentPadding 추가
